src/Analyser/TypeSpecifier.php

```diff
@@ -1102,6 +1104,10 @@ private function specifyTypesForConstantBinaryExpression(
 			&& in_array(strtolower((string) $exprNode->name), ['count', 'sizeof'], true)
 			&& $constantType instanceof ConstantIntegerType
```

Comment:
in a followup PR I would factor out the 2 `$constantType instanceof ConstantIntegerType` if-blocks of this method into the parent caller and do a proper integer range check instead, which will fix https://github.com/phpstan/phpstan/issues/11558

I don't do it now to keep this PR simpler and the next one will also be easier to review